### ISSUE TITLE
Method similar to train_test_split of sklearn

### ISSUE THREAD

**Issue by lalusaud** [yoshoku/rumale#21]
@yoshoku - Thank you for this awesome gem.

I tried to convert an example simple linear regression example from Python to Ruby. Here is my [Python example](https://github.com/lalusaud/ml/blob/master/simple_linear_regression/SimpleLinearRegression-Python.ipynb) and [Ruby example](https://github.com/lalusaud/ml/blob/master/simple_linear_regression/SimpleLinearRegression-Ruby.ipynb).

In sklearn, train_test_split seems like an easy function to convert dataset into training and test sets. 
```
# Import Dataset
dataset = pd.read_csv('salary_data.csv')
X = dataset.iloc[:, :-1].values # Take all rows and columns except last one
y = dataset.iloc[:, 1].values # Take all rows of column with index 1

# Split dataset into Training and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state = 0)
```

However, I couldn't find something similar in rumale. So I implemented like this:
```
# Import Dataset
df = Daru::DataFrame.from_csv('salary_data.csv')

# Convert dataset to Numo::NArray dataset
data = Numo::DFloat.cast df['YearsExperience', 'Salary'].to_a[0].map { |data| data.values }

# Split dataset into Training and Test set
x_train, y_train, x_test, y_test = nil
Rumale::ModelSelection::ShuffleSplit.new(test_size: 0.3, n_splits: 1, random_seed: 1).split(data).each do |train, test|
  x_train, y_train = data[train, true][true, 0..-2], data[train, true][true, 1..-1]
  x_test, y_test = data[test, true][true, 0..-2], data[test, true][true, 1..-1]
end
```
If I am not missing an easy way provided within rumale to split dataset, I think it would be great to have a method similar to train_test_split in rumale.

**Comment by yoshoku** [comment]
@lalusaud  Thank you for your suggestion and the useful notebook. I would like to implement a module function equivalent to train_test_split in the next version. 
By the way, I always obtain random split as follows:

```ruby
train_ids, test_ids = Rumale::ModelSelection::ShuffleSplit.new(
                        n_splits: 1, test_size: 0.3, random_seed: 1
                      ).split(x, y).first
x_train = x[train_ids, true]
y_train = y[train_ids]
x_test = x[test_ids, true]
y_test = y[test_ids]
```

**Comment by lalusaud** [comment]
Awesome! I look forward to the module. Thank you for the cleaner suggestion @yoshoku, I will update my notebook code.

**Comment by yoshoku** [comment]
@lalusaud I have released version 1.8.0 including train_test_split method. Please confirm that.

https://yoshoku.github.io/rumale/doc/Rumale/ModelSelection.html#train_test_split-class_method

```ruby
x_train, x_test, y_train, y_test = Rumale::ModelSelection.train_test_split(
  x, y, test_size: 0.4, random_seed: 1
)
```

**Comment by lalusaud** [comment]
Works exactly as it is supposed to! Thank you! 🙏
